Gems/Atom/RPI/Code/Source/RPI.Public/Image/StreamingImageController.cpp

```diff
@@ -128,25 +166,35 @@ namespace AZ
         {
             StreamingImageContext* context = image->m_streamingContext.get();
 
-            // Atomic min operation on target mip level. OnSetTargetMip can be called many times per frame and
-            // we want to keep the minimum (most detailed) mip level.
-            uint16_t mipLevelPrev = context->m_mipLevelTarget;
-            while (mipLevelPrev > mipLevelTarget && !context->m_mipLevelTarget.compare_exchange_weak(mipLevelPrev, mipLevelTarget));
-
+            context->m_mipLevelTarget = mipLevelTarget;
             context->m_lastAccessTimestamp = m_timestamp;
 
-            // Any time a lower mip value is requested, we need to make sure that image is queued for a reset between frames
-
-            const bool queuedForMipTargetReset = context->m_queuedForMipTargetReset.exchange(true);
-
-            if (!queuedForMipTargetReset)
+            // update image priority and re-insert the image
+            if (!context->m_queuedForMipExpand)
             {
-                m_mipTargetResetMutex.lock();
-                m_mipTargetResetQueue.emplace_back(context);
-                m_mipTargetResetMutex.unlock();
+                AZStd::lock_guard<AZStd::mutex> lock(m_imageListAccessMutex);
+                EvictUnusedMips(image);
```

Comment:
Do you need to check if this function returned true before erasing it from m_streamableImages?

Reply:
Nope. The priority changes when the target is changed regardless if there are mips evicted.